pkg/gardener/shoot/extender/provider/provider.go

```diff
@@ -313,25 +312,28 @@ func setMachineImage(provider *gardener.Provider, defMachineImgName, defMachineI
 	}
 }
 
-// HACK: can be removed when https://github.com/kyma-project/kyma-environment-broker/issues/1766 was fixed
-func correctOneNodeWorkerPools(log *logr.Logger, provider *gardener.Provider, existingWorkers []gardener.Worker) {
+func alignWorkersWithGardener(log *logr.Logger, provider *gardener.Provider, existingWorkers []gardener.Worker) {
 	existingWorkersMap := make(map[string]gardener.Worker)
 	for _, existing := range existingWorkers {
 		existingWorkersMap[existing.Name] = existing
 	}
 
 	for i := range provider.Workers {
 		alignedWorker := &provider.Workers[i]
-		if len(alignedWorker.Zones) != 1 {
-			continue
-		}
 
 		if existing, found := existingWorkersMap[alignedWorker.Name]; found {
-			if log != nil {
-				log.Info("Warning: resulting shoot doesn't contain the state defined in the Runtime CR. It is a result of a workaround for Kyma Environment Broker issue #1766 .")
+			if alignedWorker.UpdateStrategy == nil {
+				alignedWorker.UpdateStrategy = existing.UpdateStrategy
 			}
 
-			alignedWorker.Zones = existing.Zones
+			for _, zone := range existing.Zones {
```

Comment:
I think this code no longed makes sense. There was a problem in KEB with one zone clusters, but for multizoned it was working ok.

Reply:
I understand that we should relay here 100% on KEB input data here, but the consequence is that if the data is not correct Gardener will reject update call and RuntimeCR will end in failed state .